### PR TITLE
Add recent games tab

### PR DESCRIPTION
**Pull request by Luchanso**
* Fixed warnings about prop-types (my console was red before it)
* Add ErrorBox component
* Add tab "Recent" on hero page
image

@howardchung, your https://github.com/odota/core/pull/1542 branch not work on my local machine. I switched branch, added Steam API Key and wait data from Dota API, next I try fetch data "http://localhost:5000/api/heroes/1/matches?", and got empty array in response

Same problem with '/api/proMatches' and '/api/publicMatches?mmr_descending=1' endpoints

### CODE REVIEW

**Comment by howardchung** [comment]
This only counts pro matches, so you probably just don't have one of those.  You can look up a pro match with antimage via the Explorer and then request a parse of that match on your local machine

**Comment by howardchung** [comment]
Suggestions:
* Add k/d/a for each match (might need to update the query in core)
* Make an "overview" section similar to what we have for players, where we request multiple endpoints and display multiple tables

**Comment by Luchanso** [comment]
@howardchung yes, I don't have pro matches on my local michine
Okey, I will add k/d/a and overview :ok_hand:

**Comment by Luchanso** [comment]
* Added pagination
* Added K/D/A
image

Now I doing "overviews", and maybe will do other things from #1143

**Comment by Luchanso** [comment]
* Add hero attributes
* Add hero roles
image

**Comment by Luchanso** [comment]
I need some time for thinking about other data:

"icon"
"base_health"
"base_health_regen"
"base_mana"
"base_mana_regen"
"base_armor"
"base_mr"
"base_attack_min"
"base_attack_max"
"attack_range"
"projectile_speed"
"attack_rate"
"move_speed"
"turn_rate"
"cm_enabled"
"legs" 😄 👣
"pro_ban"
"hero_id"
"pro_win"
"pro_pick"
"2000_pick"
"2000_win"
"5000_pick"
"5000_win"
"1000_pick"
"1000_win"
"3000_pick"
"3000_win"
"4000_pick"
"4000_win"

**Comment by Luchanso** [comment]
In progress

**Comment by Luchanso** [comment]
* [x] Add other hero attributes
image
And now:
* [ ] I changing text on icons
* [ ] Fixing some bugs with adoptive layout

**Comment by Luchanso** [comment]
Change style and colors

image

* [ ] Fixing some bugs with layout on mobile

**Comment by Luchanso** [comment]
I found bug and fixed it. Bug about languages key:

image


Fixed version:
If key not found in selected language, then we take this key from en_US language
image

@howardchung and I'm done. Result:
image

**Comment by Luchanso** [comment]
I will add some ui for duration/players/matchups in this PR later, okey?

**Comment by howardchung** [comment]
The match list should probably include the player who played the hero in that game.

**Comment by Luchanso** [comment]
Okey, I will add it

**Comment by Luchanso** [comment]
* [x] Added column account id for recent matches
image

**Comment by Luchanso** [comment]
I'm added matchups page
image

It remains to add the /heroes/{heroId}/durations and /heroes/{heroId}/players

**Comment by howardchung** [comment]
For durations you can use the HistogramGraph visualization that's already implemented.

We probably want to add a note somewhere that this data is from pro matches only.

Maybe we should retire the benchmarks section since that's from public games.

**Comment by Luchanso** [comment]
Okey, I'll add note "* All data are taken from PRO games"

I think we just need add note in benchmarks tab "* Benchmarks data are taken from Public games"

And I don't understand "durations", can you explain me, what is it?

**Comment by howardchung** [comment]
It's the win rate of the hero for each range of 5 minutes (rounded down).

0-5 minutes, 5-10 minutes, 10-15 minutes, etc.  Basically a measure of
whether a hero is strong late game or early game.

The histogram has a bar indicating the number of games in each bucket with
a color indicating the win rate (greener = higher win %)

On Sat, Oct 14, 2017 at 12:47 PM, Loutchansky Oleg <notifications@github.com

**Comment by Luchanso** [comment]
Thank you, now I unserstand what is it.

**Comment by blukai** [comment]
@Luchanso maybe make hero page look similar to how github profile looks like?

shitty photoshop: 🦄

**Comment by Luchanso** [comment]
@blukai Okey, I will experiment with it

**Comment by Luchanso** [comment]
Finished tab "Durations"
image

Now I doing /players page and experement with layout

**Comment by Luchanso** [comment]
Done /players page
image

I will resolve conflicts and try to change layout

**Comment by Luchanso** [comment]
@blukai, @howardchung, I was changed layout on heroes page. If you like this, you can review and merge, if not, I can remove changes or I'm ready to listen to your suggestions

image

**Comment by Luchanso** [comment]
Ahh, want remove useful (duplicates) labels in tab

**Comment by Luchanso** [comment]
Done

File: src/components/Hero/AttributesBlock.jsx [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+`;
+
+const Label = styled.span`
+  margin-right: 5px;
+  color: ${constants.colorMutedLight};
+`;
+
+const renderIcon = Icon => <Icon style={{ width: 20, marginRight: 4 }} />;
+
+const calcArmorPercent = hero => Math.round(0.06 * hero / (1 + (0.06 * hero)) * 100);
```
Comment (howardchung) [inline src/components/Hero/AttributesBlock.jsx]: I'm not quite sure what this is, can you annotate/comment it?

**Comment by howardchung** [comment]
Comments:

* I think each tab should have the Heading component, and the subtitle value should be "Data from public matches" or "Data from professional matches"
* Recent matches tab should have the player's name displayed, not their account ID
* KDA isn't showing up (need to pass the data down from core?)
* How is the advantage column calculated in the Matchups tab?
* Matchups should be sorted by number of games descending (can probably be done on backend)
* Players data is from pro games, not public
* Same issue with showing the account ID instead of the name
* Players list should be sorted in descending order (can probably be done by changing the backend query)

File: src/components/Hero/AttributesBlock.jsx (hunk 1 above)
Comment (Luchanso) [reply src/components/Hero/AttributesBlock.jsx]: Yes, sure

**Comment by Luchanso** [comment]
Okay, I'll do

**Comment by Luchanso** [comment]
Tasklist:
- [x] I think each tab should have the Heading component, and the subtitle value should be "Data from public matches" or "Data from professional matches"
- [x] Recent matches tab should have the player's name displayed, not their account ID
- [ ] KDA isn't showing up (need to pass the data down from core?)
- [ ] How is the advantage column calculated in the Matchups tab?
- [ ] Matchups should be sorted by number of games descending (can probably be done on backend)
- [x] Players data is from pro games, not public
- [ ] Same issue with showing the account ID instead of the name
- [ ] Players list should be sorted in descending order (can probably be done by changing the backend query)

**Comment by Luchanso** [comment]
@howardchung anything else?

**Comment by howardchung** [comment]
Some final tweaks:
* Use the small muted font for event name on https://deploy-preview-1247--chef-sebastian-43878.netlify.com/heroes/102/recent (see /matches page for example)
* https://deploy-preview-1247--chef-sebastian-43878.netlify.com/heroes/102/matchups it's still not clear to the end user how the advantage column is calculated.  I assume this is the Wilson score?  We might want to add a tooltip explaining what this is
* https://deploy-preview-1247--chef-sebastian-43878.netlify.com/heroes/102/players use 2 decimal places for win % and remove the % (consistency with other tab)

**Comment by Luchanso** [comment]
* [x] Use the small muted font for event name on /heroes/{heroId}/recent
* [x] Tooltip for advantage column about Wilson score
* [x] Use 2 decimal places for win % and remove the % (consistency with other tab)

@howardchung, can you review again?

**Comment by howardchung** [comment]
LGTM, thanks for the contribution!

**Comment by Luchanso** [comment]
@howardchung thank you too, it was good experience. I'll try continue contribute in free time :+1:

**Comment by howardchung** [comment]
If you want to work on something similar teams pages are still available #1142

**Pull request merged**